### ISSUE TITLE
Evaluation error on master since after 2021-01-02 09:24:30

### ISSUE THREAD

**Issue by FRidh** [NixOS/nixpkgs#108304]
**Describe the bug**
Hydra fails to evaluate
https://hydra.nixos.org/jobset/nixpkgs/trunk#tabs-errors

**Comment by FRidh** [comment]
Found issue https://github.com/NixOS/nixpkgs/pull/107030#issuecomment-753591854.

**Comment by FRidh** [comment]
Fixed with 6c9b5071e4d645c8d6757175f398c52d4883eb30.

**Comment by FRidh** [comment]
> Found issue #107030 (comment).

That was only one part.

**Comment by FRidh** [comment]
I think we have an infinite recursion related to `pinpointDependencies.js`, which is nodePackages. Likely culprit ada3931e6dab2c08e4278f5ff5645a478fd73aef.

<details>
26777 <... read resumed>"stla\0\0\0\0G\0\0\0\0\0\0\0/nix/store/2xd38v4nyqn520imm9j9qsnsp42c7zx8-pinpointDependencies.js.drv\0\0\0\0\0\0\0\0\0@\0\0\0\0\0\0\0005db026328425e88eac0251477eec7f5c909d5c4b6f166b6951a95f8c45a08e86\3\0\0\0\0\0\0\0>\0\0\0\0\0\0\0/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh\0\0<\0\0\0\0\0\0\0/nix/store/qhnyj5i7w44hmhw6x5jxlzy6lnx0pwl9-bash-4.4-p23.drv\0\0\0\0<\0\0\0\0\0\0\0/nix/store/w61rvw1zfrwgvl50mfqvynh4gsjpvfhf-stdenv-linux.drv\0\0\0\0]\235\361_\0\0\0\0\0\r\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0@\0\0\0\0\0\0\0text:sha256:11bp8shvz971f6118rxhw58j6hnzs28wmxwkpcwlf3kixx5q5was", 32768) = 496
26777 madvise(0x7f9320c31000, 8368128, MADV_DONTNEED) = 0
26777 exit(0)                           = ?
26777 +++ exited with 0 +++
7872  <... futex resumed>)              = 0
7872  write(16, "\7\0\0\0\0\0\0\0\33\0\0\0\0\0\0\0pinpointDependencies.js.drv\0\0\0\0\0\v\0\0\0\0\0\0\0text:sha256\0\0\0\0\0\3\0\0\0\0\0\0\0>\0\0\0\0\0\0\0/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh\0\0<\0\0\0\0\0\0\0/nix/store/qhnyj5i7w44hmhw6x5jxlzy6lnx0pwl9-bash-4.4-p23.drv\0\0\0\0<\0\0\0\0\0\0\0/nix/store/w61rvw1zfrwgvl50mfqvynh4gsjpvfhf-stdenv-linux.drv\0\0\0\0\0\0\0\0\0\0\0\0", 304) = 304
7872  clone(child_stack=0x7f9321430cf0, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID <unfinished ...>
26778 set_robust_list(0x7f93214319e0, 24 <unfinished ...>
7872  <... clone resumed>, parent_tid=[26778], tls=0x7f9321431700, child_tidptr=0x7f93214319d0) = 26778
26778 <... set_robust_list resumed>)    = 0
26778 read(16,  <unfinished ...>
7872  write(16, "\213\f\0\0\0\0\0\0Derive([(\"out\",\"/nix/store/q632m462qglfj7yispcwb2fxdvv6bd8m-pinpointDependencies.js\",\"\",\"\")],[(\"/nix/store/qhnyj5i7w44hmhw6x5jxlzy6lnx0pwl9-bash-4.4-p23.drv\",[\"out\"]),(\"/nix/store/w61rvw1zfrwgvl50mfqvynh4gsjpvfhf-stdenv-linux.drv\",[\"out\"])],[\"/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh\"],\"aarch64-linux\",\"/nix/store/nqwkvgs0c6g35kk0zq9szwmh52w44vxx-bash-4.4-p23/bin/bash\",[\"-e\",\"/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh\"],[(\"allowSubstitutes\",\"\"),(\"buildCommand\""..., 3227) = 3227
7872  futex(0x7f93214319d0, FUTEX_WAIT, 26778, NULL <unfinished ...>
26778 <... read resumed>"stla\0\0\0\0G\0\0\0\0\0\0\0/nix/store/2xd38v4nyqn520imm9j9qsnsp42c7zx8-pinpointDependencies.js.drv\0\0\0\0\0\0\0\0\0@\0\0\0\0\0\0\0005db026328425e88eac0251477eec7f5c909d5c4b6f166b6951a95f8c45a08e86\3\0\0\0\0\0\0\0>\0\0\0\0\0\0\0/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh\0\0<\0\0\0\0\0\0\0/nix/store/qhnyj5i7w44hmhw6x5jxlzy6lnx0pwl9-bash-4.4-p23.drv\0\0\0\0<\0\0\0\0\0\0\0/nix/store/w61rvw1zfrwgvl50mfqvynh4gsjpvfhf-stdenv-linux.drv\0\0\0\0]\235\361_\0\0\0\0\0\r\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0@\0\0\0\0\0\0\0text:sha256:11bp8shvz971f6118rxhw58j6hnzs28wmxwkpcwlf3kixx5q5was", 32768) = 496
26778 madvise(0x7f9320c31000, 8368128, MADV_DONTNEED) = 0
26778 exit(0)                           = ?
26778 +++ exited with 0 +++
7872  <... futex resumed>)              = 0
7872  write(16, "\7\0\0\0\0\0\0\0\33\0\0\0\0\0\0\0pinpointDependencies.js.drv\0\0\0\0\0\v\0\0\0\0\0\0\0text:sha256\0\0\0\0\0\3\0\0\0\0\0\0\0>\0\0\0\0\0\0\0/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh\0\0<\0\0\0\0\0\0\0/nix/store/qhnyj5i7w44hmhw6x5jxlzy6lnx0pwl9-bash-4.4-p23.drv\0\0\0\0<\0\0\0\0\0\0\0/nix/store/w61rvw1zfrwgvl50mfqvynh4gsjpvfhf-stdenv-linux.drv\0\0\0\0\0\0\0\0\0\0\0\0", 304) = 304
7872  clone(child_stack=0x7f9321430cf0, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tid=[26779], tls=0x7f9321431700, child_tidptr=0x7f93214319d0) = 26779
26779 set_robust_list(0x7f93214319e0, 24) = 0
26779 read(16,  <unfinished ...>
7872  write(16, "\213\f\0\0\0\0\0\0Derive([(\"out\",\"/nix/store/q632m462qglfj7yispcwb2fxdvv6bd8m-pinpointDependencies.js\",\"\",\"\")],[(\"/nix/store/qhnyj5i7w44hmhw6x5jxlzy6lnx0pwl9-bash-4.4-p23.drv\",[\"out\"]),(\"/nix/store/w61rvw1zfrwgvl50mfqvynh4gsjpvfhf-stdenv-linux.drv\",[\"out\"])],[\"/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh\"],\"aarch64-linux\",\"/nix/store/nqwkvgs0c6g35kk0zq9szwmh52w44vxx-bash-4.4-p23/bin/bash\",[\"-e\",\"/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh\"],[(\"allowSubstitutes\",\"\"),(\"buildCommand\""..., 3227) = 3227
7872  futex(0x7f93214319d0, FUTEX_WAIT, 26779, NULL <unfinished ...>
26779 <... read resumed>"stla\0\0\0\0G\0\0\0\0\0\0\0/nix/store/2xd38v4nyqn520imm9j9qsnsp42c7zx8-pinpointDependencies.js.drv\0\0\0\0\0\0\0\0\0@\0\0\0\0\0\0\0005db026328425e88eac0251477eec7f5c909d5c4b6f166b6951a95f8c45a08e86\3\0\0\0\0\0\0\0>\0\0\0\0\0\0\0/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh\0\0<\0\0\0\0\0\0\0/nix/store/qhnyj5i7w44hmhw6x5jxlzy6lnx0pwl9-bash-4.4-p23.drv\0\0\0\0<\0\0\0\0\0\0\0/nix/store/w61rvw1zfrwgvl50mfqvynh4gsjpvfhf-stdenv-linux.drv\0\0\0\0]\235\361_\0\0\0\0\0\r\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0@\0\0\0\0\0\0\0text:sha256:11bp8shvz971f6118rxhw58j6hnzs28wmxwkpcwlf3kixx5q5was", 32768) = 496
26779 madvise(0x7f9320c31000, 8368128, MADV_DONTNEED) = 0
26779 exit(0)                           = ?
26779 +++ exited with 0 +++
7872  <... futex resumed>)              = 0
7872  mmap(NULL, 134217728, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_NORESERVE, -1, 0) = 0x7f92caf80000
7872  munmap(0x7f92caf80000, 17301504)  = 0
7872  munmap(0x7f92d0000000, 49807360)  = 0
7872  mprotect(0x7f92cc000000, 4468736, PROT_READ|PROT_WRITE) = 0
7872  --- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_ACCERR, si_addr=0x7f9308420000} ---
7872  rt_sigaction(SIGSEGV, {sa_handler=SIG_DFL, sa_mask=~[RTMIN RT_1], sa_flags=SA_RESTORER, sa_restorer=0x7f932a3ffee0}, NULL, 8) = 0
7872  rt_sigreturn({mask=[HUP INT PIPE TERM WINCH]}) = 140269475463170
7872  --- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_ACCERR, si_addr=0x7f9308420000} ---
7872  +++ killed by SIGSEGV (core dumped) +++
18952 <... read resumed>"", 1)          = 0
18952 --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_DUMPED, si_pid=7872, si_uid=1000, si_status=SIGSEGV, si_utime=337, si_stime=245} ---
18952 close(4)                          = 0
18952 close(8)                          = 0
18952 madvise(0x7f9308420000, 8368128, MADV_DONTNEED) = 0
18952 exit(0)                           = ?
18935 <... futex resumed>)              = 0
18952 +++ exited with 0 +++
18935 mmap(NULL, 1662976, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f930828a000
18935 mmap(NULL, 3321856, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f92fbcd5000
18935 munmap(0x7f930828a000, 1662976)   = 0
18935 munmap(0x7f92fbcd5000, 3321856)   = 0
18935 write(2, "\33[31;1merror:\33[0m\33[34;1m --- EndOfFile ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- hydra-eval-jobs\33[0m\nunexpected EOF reading a line\n", 295) = 295
18935 exit_group(1)                     = ?
18951 <... futex resumed>)              = ?
18950 <... futex resumed>)              = ?
18949 <... futex resumed>)              = ?
18948 <... futex resumed>)              = ?
18947 <... futex resumed>)              = ?
18946 <... futex resumed>)              = ?
18945 <... futex resumed>)              = ?
18944 <... futex resumed>)              = ?
18943 <... futex resumed>)              = ?
18942 <... futex resumed>)              = ?
18941 <... futex resumed>)              = ?
18940 <... futex resumed>)              = ?
18939 <... futex resumed>)              = ?
18938 <... futex resumed>)              = ?
18937 <... futex resumed>)              = ?
18936 <... rt_sigtimedwait resumed> <unfinished ...>) = ?
18951 +++ exited with 1 +++
18950 +++ exited with 1 +++
18949 +++ exited with 1 +++
18948 +++ exited with 1 +++
18947 +++ exited with 1 +++
18946 +++ exited with 1 +++
18945 +++ exited with 1 +++
18944 +++ exited with 1 +++
18943 +++ exited with 1 +++
18942 +++ exited with 1 +++
18941 +++ exited with 1 +++
18940 +++ exited with 1 +++
18939 +++ exited with 1 +++
18938 +++ exited with 1 +++
18937 +++ exited with 1 +++
18936 +++ exited with 1 +++
18935 +++ exited with 1 +++

</details>

**Comment by FRidh** [comment]
New trace!
```
error: --- EvalError --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix-env
at: (253:36) in file: /home/freddy/Code/libraries/nixpkgs_2/pkgs/os-specific/darwin/apple-sdk/default.nix

   252|     AppKit = stdenv.lib.overrideDerivation super.AppKit (drv: {
   253|       __propagatedImpureHostDeps = drv.__propagatedImpureHostDeps ++ [
      |                                    ^
   254|         "/System/Library/PrivateFrameworks/"

attribute '__propagatedImpureHostDeps' missing
------------------------------------------------------------------------------------------------------------------- show-trace -------------------------------------------------------------------------------------------------------------------
trace: while evaluating the attribute '__propagatedImpureHostDeps' of the derivation 'apple-framework-AppKit'
at: (68:5) in file: /home/freddy/Code/libraries/nixpkgs_2/pkgs/os-specific/darwin/apple-sdk/default.nix

    67|   framework = name: deps: stdenv.mkDerivation {
    68|     name = "apple-framework-${name}";
      |     ^
    69| 

trace: while evaluating 'escapeShellArg'
at: (318:20) in file: /home/freddy/Code/libraries/nixpkgs_2/lib/strings.nix

   317|   */
   318|   escapeShellArg = arg: "'${replaceStrings ["'"] ["'\\''"] (toString arg)}'";
      |                    ^
   319| 

trace: from call site
trace: while evaluating 'concatMapStringsSep'
at: (110:5) in file: /home/freddy/Code/libraries/nixpkgs_2/lib/strings.nix

   109|     # List of input strings
   110|     list: concatStringsSep sep (map f list);
      |     ^
   111| 

trace: from call site
at: (65:6) in file: /home/freddy/Code/libraries/nixpkgs_2/pkgs/os-specific/darwin/apple-sdk/default.nix

    64|     substArgs = lib.concatMap (x: [ "--subst-var-by" x deps'."${x}" ]) (lib.attrNames deps');
    65|   in lib.escapeShellArgs substArgs;
      |      ^
    66| 

trace: while evaluating 'mkFrameworkSubs'
at: (61:27) in file: /home/freddy/Code/libraries/nixpkgs_2/pkgs/os-specific/darwin/apple-sdk/default.nix

    60| 
    61|   mkFrameworkSubs = name: deps:
      |                           ^
    62|   let

trace: from call site
at: (157:36) in file: /home/freddy/Code/libraries/nixpkgs_2/pkgs/os-specific/darwin/apple-sdk/default.nix

   156|         echo "Fixing re-exports in $tbd"
   157|         substituteInPlace "$tbd" ${mkFrameworkSubs name deps}
      |                                    ^
   158| 

trace: while evaluating the attribute 'installPhase' of the derivation 'apple-framework-Cocoa'
at: (68:5) in file: /home/freddy/Code/libraries/nixpkgs_2/pkgs/os-specific/darwin/apple-sdk/default.nix

    67|   framework = name: deps: stdenv.mkDerivation {
    68|     name = "apple-framework-${name}";
      |     ^
    69| 

trace: while evaluating the attribute 'buildInputs' of the derivation 'chunkwm-0.4.9'
at: (192:11) in file: /home/freddy/Code/libraries/nixpkgs_2/pkgs/stdenv/generic/make-derivation.nix

   191|         // (lib.optionalAttrs (!(attrs ? name) && attrs ? pname && attrs ? version)) {
   192|           name = "${attrs.pname}-${attrs.version}";
      |           ^
   193|         } // (lib.optionalAttrs (stdenv.hostPlatform != stdenv.buildPlatform && !dontAddHostSuffix && (attrs ? name || (attrs ? pname && attrs ? version)))) {

trace: while querying the derivation named 'chunkwm-0.4.9'
```

**Comment by FRidh** [comment]
Found it! ae64bc63d564fda329b963551085cbbde342b561

**Comment by FRidh** [comment]
Fixed with aaf4dddc1d557457e5f405c3cb6d2d5ea12d1068.
